Updated the FIPS cryptographic algorithms tests

patrikcerbak opened the pull request:
There are a few changes to the FIPS tests with this update:
- the `CheckAlgorithms.java` file is now JDK8 compatible, the compilation shouldn't be a problem anymore
- now, the FIPS tests are all skipped by default unless a `OTOOL_CONTAINERQA_RUNCRYPTO` variable is set to `true`
- deleted the tests that were setting `update-crypto-policies --set FIPS` in the container as, after discussion, they were deemed unnecessary
- added `isHostAndContainerCryptoIdentical` test, which for now just prints crypto policies on the host system and in the container
- added `doesManuallySettingFipsCrash` test, which for now just prints the container's exit code when trying to set `update-crypto-policies --set FIPS` inside (it returns zero everywhere, except on FIPS-enabled RHELs)

As for the testing, I tried to run them in a VM on all of these combinations, and the tests passed everywhere (there was no crash):

| **Host OS** | **FIPS?** | **Container** |
|:-----------:|:---------:|:-------------:|
|   RHEL 8    |     no    |   ubi8-jdk8   |
|             |           |   ubi8-jdk21  |
|             |    yes    |   ubi8-jdk8   |
|             |           |   ubi8-jdk21  |
|  Fedora 40  |     no    |   ubi8-jdk21  |
|             |    yes    |   ubi8-jdk21  |

All of the new FIPS tests just "print" the results, no real testing is done. In the future, I can edit them to make some actual assertions.

sefroberg commented on containersQa/testlib.bash:
```diff
@@ -930,40 +931,42 @@ EOF
   cipherListCode=`cat $LIBCQA_SCRIPT_DIR/CipherList.java`
 }
 
+function isHostAndContainerCryptoIdentical() {
```
Your method is asking a question you are not answering. Either add a check to the method or change the method name to logHostAndContainerCryptoPolicy()

judovana replied:
It is true, that now it is just logging. But on the longer run it will be asserting.  thus I vote to keep  isHostAndContainerCryptoIdentical.  However. the test itself is `..._isHostAndContainerCryptoIdentical` thus the method indeed can be `logHostAndContainerCryptoPolicy` and will jsut logging, and the new method, in far future will be `isHostAndContainerCryptoPolicy` and that one will be added, and will be asserting. So whether to rename or not, depends on future, whetehr the current method will be also asserting or not

patrikcerbak replied:
I changed the method name and added some more complex logging.

sefroberg commented on containersQa/testlib.bash:
```diff
@@ -930,40 +931,42 @@ EOF
+  echo "The crypto policy in container: $containerCryptoPolicy"
+}
+
+function doesManuallySettingFipsCrash() {
```
I suggest changing the method to validateManualSettingFIPSWithNoCrash()

patrikcerbak replied:
I changed the method name and added some more complex logging.

sefroberg commented on containersQa/testlib.bash:
```diff
@@ -930,40 +931,42 @@ EOF
+  containerReturnCode="$?"
+  set -e
+
+  echo "The container returned $containerReturnCode as its exit code when manually setting FIPS."
```
Rather than just logging the return value, apply it to an if statement so this becomes a test and not just a logging event.

judovana replied:
yup. Pls be clear what it returned.

sefroberg reviewed (dismissed):
Good start. I think you can update some of your existing code to make the methods full tests rather than just logging output. Please also run this on Rhel 8, Rhel 9 and against JDK 8, 11, and 21. I would be very surprised if it passes in those three and there be an issue we miss in JDK 17.

patrikcerbak commented:
I updated the new test names to better match what they do. I also updated the `echo`s to do some more complex logging (with if statements). When properly tested, I thinks that the logging can be changed to asserting.

I tried it in a VM on all of these variants and the results were as expected:

| **Host OS** | **FIPS?** | **Container** |
|:-----------:|:---------:|:-------------:|
|    RHEL 8   |     no    |   ubi8-jdk8   |
|             |           |   ubi8-jdk21  |
|             |           |   ubi9-jdk11  |
|             |    yes    |   ubi8-jdk8   |
|             |           |   ubi8-jdk21  |
|             |           |   ubi9-jdk11  |
|    RHEL 9   |     no    |   ubi8-jdk8   |
|             |           |   ubi8-jdk21  |
|             |           |   ubi9-jdk11  |
|             |    yes    |   ubi8-jdk8   |
|             |           |   ubi8-jdk21  |
|             |           |   ubi9-jdk11  |
|  Fedora 41  |     no    |   ubi8-jdk8   |
|             |           |   ubi8-jdk21  |
|             |           |   ubi9-jdk11  |
|             |    yes    |   ubi8-jdk8   |
|             |           |   ubi8-jdk21  |
|             |           |   ubi9-jdk11  |
